pkg/serving/config_changes.go

```diff
@@ -374,31 +374,34 @@ func UpdateResources(template *servingv1.RevisionTemplateSpec, requestsResourceL
 	return nil
 }
 
-// ServiceOnlyLabels should only appear on the Service and NOT on the
-// Revision template
-var ServiceOnlyLabels = map[string]struct{}{
-	serving.GroupName + "/visibility": {},
-}
+// // ServiceOnlyLabels should only appear on the Service and NOT on the
+// // Revision template
+// var ServiceOnlyLabels = map[string]struct{}{
+// 	serving.GroupName + "/visibility": {},
+// }
 
 // UpdateLabels updates the labels identically on a service and template.
 // Does not overwrite the entire Labels field, only makes the requested updates
-func UpdateLabels(service *servingv1.Service, template *servingv1.RevisionTemplateSpec, toUpdate map[string]string, toRemove []string) error {
+func UpdateLabels(service *servingv1.Service, template *servingv1.RevisionTemplateSpec,
```

Comment:
Having functions with more than 4 or 5 arguments is a code smell, especially when the have the same types. This is not good API design and confuses users and code readers. Instead, its better to either split up a 'god' function (i.e. in a simpler method which might then be called twice) or introduce an _option struct_ (which allows for "named" parameters (field of the structure) in favor of positional arguments). If possible the formere solution is easier to implement and more lightweight. The latter (option struct) should be used only if it can't be avoid. 

This seems kind of a nit-pick comment for now, but maintaining god methods that do too much is really hard. Also using smaller method can help in guaranteeing consistency.

Reply:
E.g. one could also use two methods `UpdateServiceLabels` or `UpdateRevisionTemplateLabels` (or maybe also `UpdatePodLabels` as this is where the lable will end up and might be easier to understand).

I think this is the better option.